### PR TITLE
update proto https://community.libra.org/t/testnet-update-scheduled-f…

### PR DESCRIPTION
**Pull request by wenshenjun**
update proto https://community.libra.org/t/testnet-update-scheduled-for-july-10th-with-breaking-changes/1143

### CODE REVIEW

**Comment by totiz** [comment]
I tested, It had errors about type changing String -> Number,
So in file lib/client/Encoder.ts need to adjust from this.
```javascript
rawTransaction.setExpirationTime(transaction.expirationTime.**toString(10)**);
rawTransaction.setGasUnitPrice(transaction.gasContraint.gasUnitPrice.**toString(10)**);
rawTransaction.setMaxGasAmount(transaction.gasContraint.maxGasAmount.**toString(10)**);
rawTransaction.setSequenceNumber(sequenceNumber.**toString(10)**);
```
to
```javascript
rawTransaction.setExpirationTime(transaction.expirationTime.**toNumber()**);
rawTransaction.setGasUnitPrice(transaction.gasContraint.gasUnitPrice.**toNumber()**);
rawTransaction.setMaxGasAmount(transaction.gasContraint.maxGasAmount.**toNumber()**);
rawTransaction.setSequenceNumber(sequenceNumber.**toNumber()**);
```

And I've successfully tested on transfer.

**Comment by wenshenjun** [comment]
@totiz  Thank you for you test.Is my wrong. I just run `npm run test`.

File: test/utils.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import diff from 'jest-diff';
+import * as diff from 'jest-diff';
```
Comment (perfectmak) [inline test/utils.ts]: What is the purpose of this change? Also, I'm getting this error when trying to run the tes, but If I revert this change, It works. Could you help out with this?

This is a screenshot of the error
Screenshot 2019-07-12 at 12 00 35 PM

File: lib/transaction/Transactions.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import BigNumber from 'bignumber.js';
+import { AdmissionControlStatus } from '../__generated__/admission_control_pb';
```
Comment (perfectmak) [inline lib/transaction/Transactions.ts]: Please can you remove this import from here. I don't think it is used and as a principle I want to limit usage of the `__generated__` classes to only the `client` module.

File: lib/client/index.ts [hunk 3]
```diff
@@ -280,8 +286,10 @@ export class LibraClient {
           new LibraTransactionResponse(
             new LibraSignedTransaction(transaction, sender.keyPair.getPublicKey(), senderSignature),
             response.getValidatorId_asU8(),
-            response.getAcStatus(),
-            response.getMempoolStatus(),
+            response.getAcStatus() === undefined ? -1 : (response.getAcStatus() as AdmissionControlStatus).getCode(),
```
Comment (perfectmak) [inline lib/client/index.ts]: Instead of `response.getActStatus() === undefined` use `response.hasAcStatus()`,
Then update the `LibraAdmissionControlStatus` enum to have a -1 value to account for the -1 default value. The name of the -1 value can be `UNKNOWN`.

File: lib/client/index.ts [hunk 4]
```diff
@@ -280,8 +286,10 @@ export class LibraClient {
...
+            response.getMempoolStatus() === undefined
```
Comment (perfectmak) [inline lib/client/index.ts]: Please also change `response.getMempoolStatus()` as explained in the comment above and update the `LibraMempoolTransactionStatus` enum to account for the -1 value.

File: test/wallet.test.ts [hunk 5]
```diff
@@ -10,7 +10,7 @@ describe('LibraWallet', () => {
         mnemonics.push(mnemonic);
       }
 
-      expect([...new Set(mnemonics)].length).toBe(100);
+      expect([...Array.from(new Set(mnemonics))].length).toBe(100);
```
[the remark this answers is not in the document]
Comment (wenshenjun) [reply test/wallet.test.ts]: image
Yes, they are the same, I just let my IDE not warn.
I using IDEA.

File: lib/transaction/Transactions.ts (hunk 2 above)
Comment (wenshenjun) [reply lib/transaction/Transactions.ts]: Ok.

File: lib/client/index.ts (hunk 3 above)
Comment (wenshenjun) [reply lib/client/index.ts]: Ok,I will do it.

File: lib/client/index.ts (hunk 4 above)
Comment (wenshenjun) [reply lib/client/index.ts]: Ok.

File: test/utils.ts (hunk 1 above)
Comment (wenshenjun) [reply test/utils.ts]: image
Because this,I will revert it. 
But i can run with `import * as diff from 'jest-diff';`
